tests/integ_tests/ephemeral_server_tests.rs

```diff
@@ -0,0 +1,138 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+use temporal_client::{TestService, WorkflowService};
+use temporal_sdk_core::{
+    ClientOptions, ClientOptionsBuilder, EphemeralExe, EphemeralExeVersion, EphemeralServer,
+    TemporaliteConfigBuilder, TestServerConfigBuilder,
+};
+use temporal_sdk_core_protos::temporal::api::workflowservice::v1::DescribeNamespaceRequest;
+use temporal_sdk_core_test_utils::NAMESPACE;
+use url::Url;
+
+#[tokio::test]
+async fn temporalite_default() {
+    let config = TemporaliteConfigBuilder::default()
+        .exe(default_cached_download())
+        .build()
+        .unwrap();
+    let server = config.start_server().await.unwrap();
+    assert_ephemeral_server(&server).await;
+}
+
+#[tokio::test]
+async fn temporalite_fixed() {
+    let config = TemporaliteConfigBuilder::default()
+        .exe(fixed_cached_download("v0.1.1"))
+        .build()
+        .unwrap();
+    let server = config.start_server().await.unwrap();
+    assert_ephemeral_server(&server).await;
+}
+
+#[tokio::test]
+async fn temporalite_shutdown_port_reuse() {
+    // Start, test shutdown, do again immediately on same port to ensure we can
+    // reuse after shutdown
+    let config = TemporaliteConfigBuilder::default()
+        .exe(default_cached_download())
+        .port(Some(10123))
```

Comment:
nit: In rare cases this port will already be taken, allocate a free port instead

Reply:
I didn't really want to expose that API or copy it over. Not too concerned here.

Reply:
It should rarely matter but eventually we'll hit this, IMHO the effort of allocating a random port is low enough to make it worth addressing this.

Leaving it up to you.

Reply:
The goal of the test is to confirm I can create a server on the same port right after I shutdown the previous one. Basically it confirms that after `shutdown` is returned, the port is still available. So I need the same port for both, but technically I could use a "free port" helper if I had to.